### PR TITLE
Review for merge with master

### PR DESCRIPTION
**Pull request by aalexren**
Check it please to merge with master 😺

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+text2.txt
+cmake-build-debug/
```
Comment (l1va) [inline .gitignore:2]: like

File: HW1/main.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+string read_text(const string& filename);
+unsigned int find_delimiter(const string& text);
+void withdraw_words(const string& text, vector<string>& _owords);
+void get_sentences(const string& text, char del, vector<string>& _osentences);
+void strip(string& s);
+vector<vector<string>> make_uppercase(const vector<string>& sentences, const vector<string>& words);
+
+int main() {
+   string full_text = read_text("../input.txt"); // without indicated words
+   vector<string> words; withdraw_words(full_text, words);
```
Comment (l1va) [inline HW1/main.cpp:30]: no comments)

File: HW1/main.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      }
+   }
+   output.close();
+
+   return 0;
+}
+
+// read from the end of the text
+unsigned int find_delimiter(const string& text) {
+   for (unsigned int i = text.length() - 1; i >= 0;) {
```
Comment (l1va) [inline HW1/main.cpp:56]: what if digit inside the text or word?

File: HW1/main.cpp [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+vector<vector<string>> make_uppercase(const vector<string>& sentences, const vector<string>& words) {
+   vector<vector<string>> changed(words.size()); // list of sentences for every word
+
+   for (int i = 0; i < words.size(); ++i) {
+      for (int j = 0; j < sentences.size(); ++j) {
+         string qwe = sentences[j];
+         auto it = qwe.find(words[i]);
+         bool add = false;
+         do {
+            if (it > 0 && isalpha(qwe[it-1])) goto crutch; // part of another word
```
Comment (l1va) [inline HW1/main.cpp:132]: overengineering

**Review by l1va — APPROVED** [review]

File: HW1/main.cpp (hunk 3 above)
Comment (aalexren) [reply HW1/main.cpp:56]: yeah, my fault, but I thought it's up to me :)

**Pull request merged**